src/getNativeComponent.tsx

```diff
@@ -0,0 +1,86 @@
+import {NativeModule} from 'react-native';
+import {
+  find,
+  NATIVE_COMPONENTS,
+  UNSUPPORTED_PAYMENT_METHODS,
+} from './ComponentMap';
+import {
+  UNKNOWN_PAYMENT_METHOD_ERROR,
+  UNSUPPORTED_PAYMENT_METHOD_ERROR,
+} from './Core/constants';
+import {PaymentMethod, PaymentMethodsResponse} from './Core/types';
+import {AdyenNativeComponentWrapper} from './AdyenNativeComponentWrapper';
+import {
+  AdyenActionComponent,
+  AdyenDropIn,
+  AdyenApplePay,
+  AdyenGooglePay,
+  AdyenInstant,
+} from './AdyenNativeModules';
+
+/**
+ * Get native component capable of handling provided payment method type.
+ */
+export function getNativeComponent(
+  typeName: string,
+  paymentMethods: PaymentMethodsResponse,
+): {
+  nativeComponent: AdyenActionComponent & NativeModule;
+  paymentMethod: PaymentMethod | undefined;
+} {
+  switch (typeName) {
+    case 'dropin':
+    case 'dropIn':
+    case 'drop-in':
+    case 'adyendropin':
+      return {
+        nativeComponent: new AdyenNativeComponentWrapper({
+          nativeModule: AdyenDropIn,
+        }),
+        paymentMethod: undefined,
```

Comment:
If we define `paymentMethod?: PaymentMethod;` as optional, we don't need to return `paymentMethod: undefined`. Same for other cases.
So it would be 
```
return {
        nativeComponent: new AdyenNativeComponentWrapper({
          nativeModule: AdyenDropIn,
        })
      };
```